File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxlen=0;

        for(int i=0;i<s.size();i++){
            unordered_map<char,int>mp;
            for(int j=i;j<s.size();j++){
                if(mp[s[j]]==1)break;
                int len=j-i+1;
                maxlen=max(maxlen,len);
                mp[s[j]]=1;
            }
        }
        return maxlen;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashMap<Character,Integer>mpp=new HashMap<>();
        int left=0;int right=0;
        int len=0;
        while(right<s.length()){
            if(mpp.containsKey(s.charAt(right)))
                left=Math.max(mpp.get(s.charAt(right))+1,left);
            //update the latest right index of the character
            mpp.put(s.charAt(right),right);
            len=Math.max(len,right-left+1);
            right++;
        }
        return len;
    }
}